pgmpy/estimators/TreeSearch.py

```diff
@@ -100,87 +96,134 @@ def estimate(self):
+            D = TreeSearch.chow_liu(
+                df_features, self.root_node, edge_weights, self.n_jobs, show_progress
+            )
+
+            # Step 3.3: Add edges from class_node to all other nodes.
+            D.add_edges_from([(class_node, node) for node in df_features.columns])
+
+            return D
+
+    @staticmethod
+    def chow_liu(
+        data, root_node, edge_weights="mutual_info", n_jobs=-1, show_progress=True
+    ):
+        """
+        Chow-Liu algorithm for estimating tree structure from given data. Refer to 
+        pgmpy.estimators.TreeSearch for more details.
+
+        Parameters
+        ----------
+        data: pandas.DataFrame object
+            dataframe object where each column represents one variable.
+
+        root_node: str, int, or any hashable python object.
+            The root node of the tree structure.
+
 
-            # add edge from class node to other nodes
-            for node in df_features.columns:
-                D.add_edge(class_node, node)
+        n_jobs: int (default: -1)
+            Number of jobs to run in parallel. `-1` means use all processors.
 
-            return DAG(D)
+        edge_weights: str or function (default: mutual info)
+            Method to use for computing edge weights. Options are:
+                1. 'mutual_info': Mutual Information Score.
+                2. 'adjusted_mutual_info': Adjusted Mutual Information Score.
+                3. 'normalized_mutual_info': Normalized Mutual Information Score.
+                4. function(array[n_samples,], array[n_samples,]): Custom function.
 
-    def chow_liu(self, data, root_node, n_jobs):
-        # create a fully connected graph over the independent variables
-        def calculate_mi(u_index, v_index):
-            return (
-                u_index,
-                v_index,
-                mutual_info_score(data.iloc[:, u_index], data.iloc[:, v_index]),
+        show_progress: boolean
+            If True, shows a progress bar for the running algorithm.
+
+        Returns
+        -------
+        model: `pgmpy.base.DAG` instance
+            The estimated model structure.
+
+        Examples
+        --------
+        >>> import numpy as np
+        >>> import pandas as pd
+        >>> from pgmpy.estimators import TreeSearch
+        >>> values = pd.DataFrame(np.random.randint(low=0, high=2, size=(1000, 5)),
+        ...                       columns=['A', 'B', 'C', 'D', 'E'])
+        >>> est = TreeSearch(values, root_node='B')
+        >>> model = est.estimate(estimator_type='chow-liu')
+        """
+        # Step 0: Check for edge weight computation method
+        if edge_weights == "mutual_info":
+            edge_weights = mutual_info_score
+        elif not isinstance(edge_weights, callable):
+            raise ValueError(
+                "edge_weights either needs to be 'mutual_info' or a function of form fun(array, array)"
```

Comment:
Going by the docstring, should the error msg include the options 'adjusted_mutual_info' and 'normalized_mutual_info'?

Reply:
Fixed.